components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,276 @@
+import {
+  InstantSearch,
+  Pagination
+} from "@alexjball/react-instantsearch-hooks-web"
+import { collection, query, where, getDocs } from "firebase/firestore"
+import ErrorPage from "next/error"
+import { useCallback, useEffect, useState } from "react"
+import TypesenseInstantSearchAdapter from "typesense-instantsearch-adapter"
+import { useMediaQuery } from "usehooks-ts"
+import { useAuth } from "../auth"
+import { Col, Row, Spinner } from "../bootstrap"
+import { usePublicProfile } from "../db"
+import { firestore } from "../firebase"
+import {
+  Header,
+  HeaderTitle,
+  StyledContainer
+} from "./StyledNewsfeedComponents"
+import { AlertCard } from "components/AlertCard/AlertCard"
+import { getServerConfig } from "components/search/common"
+import { SearchContainer } from "components/search/SearchContainer"
+import { SearchErrorBoundary } from "components/search/SearchErrorBoundary"
+import { initialSortByValue } from "components/search/SortBy"
+import { currentGeneralCourt } from "functions/src/shared"
+
+type ElementProps = {
+  bodyText: string
+  court: string
+  header: string
+  id: string
+  subheader: string
+  timestamp: string
+  topicName: string
+  type: string
+}
+
+type Props = ElementProps[]
+
+const searchClient = new TypesenseInstantSearchAdapter({
+  server: getServerConfig(),
+  additionalSearchParameters: {
+    query_by: "number,title,body",
+    exclude_fields: "body"
+  }
+}).searchClient
+
+export default function Newsfeed() {
+  return (
+    <SearchErrorBoundary>
+      <InstantSearch
+        indexName={initialSortByValue}
+        initialUiState={{
+          [initialSortByValue]: {
+            refinementList: { court: [String(currentGeneralCourt)] }
+          }
+        }}
+        searchClient={searchClient}
+        routing={false}
+        // routing={useRouting()}
+      >
+        <Layout />
+      </InstantSearch>
+    </SearchErrorBoundary>
+  )
+}
+
+const Layout = () => {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const { result: profile, loading } = usePublicProfile(user?.uid)
+  const uid = user?.uid
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
+  )
+
+  const [orgFilter, setOrgFilter] = useState<boolean>(true)
+  const [billFilter, setBillFilter] = useState<boolean>(true)
+
+  const [allResults, setAllResults] = useState<Props>([])
+  const [orgResults, setOrgResults] = useState<Props>([])
+  const [billResults, setBillResults] = useState<Props>([])
+
+  const [notificationsDisplay, setNotificationsDisplay] = useState<Props>([])
+  const [noDisplay, setNoDisplay] = useState<boolean>(true)
+
+  const onFilterChange = (trolley: string, boxType: string) => {
+    const Box = document.getElementById(boxType) as HTMLInputElement
+    if (Box?.checked) {
+      trolley ? setOrgFilter(true) : setBillFilter(true)
+    } else {
+      trolley ? setOrgFilter(false) : setBillFilter(false)
+    }
+  }
```

Comment:
Generally I'm a bit wary when I see query selectors (e.g. `document.getElementById`) in React code...  it looks like in this case we're trying to ascertain the current state of a particular input element, however I think we have access to the current state of both of the input elements that you eventually use this function on...

Perhaps we could use [a traditional approach](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable) to this problem, which might help simplify this `onFilterChange` method.

```suggestion
  const onBillFilterChange = (isFiltering: boolean) => {
    setIsFilteringBills(isFiltering)
  }
  const onOrgFilterChange = (isFiltering: boolean) => {
    setIsFilteringOrgs(isFiltering)
  }
```

It might seem tempting to combine these functions, however I think the clarity of keeping them separate helps the code readability and structure.